## Task M2.2

In [46]:
import pandas as pd
import numpy as np

#### Reading the super_df.csv

In [47]:
df = pd.read_csv('../../../../data/super_df.csv')


In [48]:
df.shape

(2527, 542)

In [49]:
df.head()

County State  County Code  Population  Deaths  Norm_Deaths  \
0  Abbeville County, SC    SC        45001      535389      38          8.0   
1     Acadia Parish, LA    LA        22001     1279727     237         19.0   
2   Accomack County, VA    VA        51001      726189      73         11.0   
3        Ada County, ID    ID        16001     8083452     988         13.0   
4      Adair County, KY    KY        21001      387950      43         12.0   

   State FIPS Code  County FIPS Code  5-digit FIPS Code State Abbreviation  \
0               45                 1              45001                 SC   
1               22                 1              22001                 LA   
2               51                 1              51001                 VA   
3               16                 1              16001                 ID   
4               21                 1              21001                 KY   

   ... Male population 65+ raw value  Total male population raw value  \
0  ...                           NaN                              NaN   
1  ...                           NaN                              NaN   
2  ...                           NaN                              NaN   
3  ...                           NaN                              NaN   
4  ...                           NaN                              NaN   

   Female population 0-17 raw value  Female population 18-44 raw value  \
0                               NaN                                NaN   
1                               NaN                                NaN   
2                               NaN                                NaN   
3                               NaN                                NaN   
4                               NaN                                NaN   

   Female population 45-64 raw value  Female population 65+ raw value  \
0                                NaN                              NaN   
1                                NaN                              NaN   
2                                NaN                              NaN   
3                                NaN                              NaN   
4                                NaN                              NaN   

   Total female population raw value  Population growth raw value   FIPS  \
0                                NaN                          NaN  45001   
1                                NaN                          NaN  22001   
2                                NaN                          NaN  51001   
3                                NaN                          NaN  16001   
4                                NaN                          NaN  21001   

   Opiod_Dispensing_Rate  
0                   32.4  
1                   32.2  
2                   19.4  
3                   60.8  
4                   71.3  

[5 rows x 542 columns]

#### Select columns that contain "raw" in their header in addition to "Norm_Deaths" as the most important variable.


In [50]:
select = [item for item in df.columns if 'raw' in item] + ['Norm_Deaths']


#### Calculate correlation coefficients between all selected columns and select those calculated for "Norm_Deaths" column.


In [51]:
ndf = np.abs(df[select].corr()['Norm_Deaths'])


#### Drop nan values from correlation matrix.


In [52]:
ndf = ndf.dropna()


#### Sort correlation coefficients and save it as a csv file.


In [53]:
ndf.sort_values().to_csv('important_socioeconomic_factors_preselected.csv')

#### Columns with correlation coefficients higher than 0.3.

In [54]:
ndf[ndf > 0.3][:10]

Premature death raw value                     0.335419
Poor mental health days raw value             0.353252
Injury deaths raw value                       0.475760
Life expectancy raw value                     0.300341
Premature age-adjusted mortality raw value    0.317211
Drug overdose deaths raw value                0.789955
Communicable disease raw value                0.542540
Cancer incidence raw value                    0.470699
Drug arrests raw value                        0.559021
Opioid hospital visits raw value              0.490949
Name: Norm_Deaths, dtype: float64

#### dtype, min and max of selected columns

In [55]:
df[['Reading proficiency raw value',
'Life expectancy raw value',
'Older adults living alone raw value',
'Premature death raw value',
'Injury hospitalizations raw value',
'Cancer incidence raw value',
'Hate crimes raw value',
'Total male population raw value',
'Communicable disease raw value',
'Drug arrests raw value'
]].dtypes

Reading proficiency raw value          float64
Life expectancy raw value              float64
Older adults living alone raw value    float64
Premature death raw value              float64
Injury hospitalizations raw value      float64
Cancer incidence raw value             float64
Hate crimes raw value                  float64
Total male population raw value        float64
Communicable disease raw value         float64
Drug arrests raw value                 float64
dtype: object

In [56]:
df[['Reading proficiency raw value',
'Life expectancy raw value',
'Older adults living alone raw value',
'Premature death raw value',
'Injury hospitalizations raw value',
'Cancer incidence raw value',
'Hate crimes raw value',
'Total male population raw value',
'Communicable disease raw value',
'Drug arrests raw value'
]].min()

Reading proficiency raw value             0.111110
Life expectancy raw value                66.883914
Older adults living alone raw value       0.231687
Premature death raw value              2610.690433
Injury hospitalizations raw value       174.300000
Cancer incidence raw value              330.300000
Hate crimes raw value                     0.384023
Total male population raw value        2176.000000
Communicable disease raw value          425.314152
Drug arrests raw value                    4.000000
dtype: float64

In [57]:
df[['Reading proficiency raw value',
'Life expectancy raw value',
'Older adults living alone raw value',
'Premature death raw value',
'Injury hospitalizations raw value',
'Cancer incidence raw value',
'Hate crimes raw value',
'Total male population raw value',
'Communicable disease raw value',
'Drug arrests raw value'
]].max()

Reading proficiency raw value               0.732910
Life expectancy raw value                  94.659083
Older adults living alone raw value         0.345087
Premature death raw value               22123.700270
Injury hospitalizations raw value         857.500000
Cancer incidence raw value                584.300000
Hate crimes raw value                       7.078644
Total male population raw value        461457.000000
Communicable disease raw value           2053.394134
Drug arrests raw value                   4489.000000
dtype: float64

## Task 3 

In [60]:
Mortality_Rate = pd.read_csv("../../../../data/Underlying Cause of Death-County-2019.txt",delimiter='\t')
Mortality_Rate

Notes                 County  County Code  \
0       NaN     Autauga County, AL         1001   
1       NaN     Autauga County, AL         1001   
2       NaN     Baldwin County, AL         1003   
3       NaN     Baldwin County, AL         1003   
4       NaN     Baldwin County, AL         1003   
...     ...                    ...          ...   
5536    NaN  Sweetwater County, WY        56037   
5537    NaN       Teton County, WY        56039   
5538    NaN       Uinta County, WY        56041   
5539    NaN       Uinta County, WY        56041   
5540    NaN    Washakie County, WY        56043   

                             Drug/Alcohol Induced Cause  \
0     Drug poisonings (overdose) Unintentional (X40-...   
1          Drug poisonings (overdose) Suicide (X60-X64)   
2     Drug poisonings (overdose) Unintentional (X40-...   
3          Drug poisonings (overdose) Suicide (X60-X64)   
4     Drug poisonings (overdose) Undetermined (Y10-Y14)   
...                                                 ...   
5536       Drug poisonings (overdose) Suicide (X60-X64)   
5537  Drug poisonings (overdose) Unintentional (X40-...   
5538  Drug poisonings (overdose) Unintentional (X40-...   
5539  Drug poisonings (overdose) Undetermined (Y10-Y14)   
5540  Drug poisonings (overdose) Unintentional (X40-...   

     Drug/Alcohol Induced Cause Code  Deaths  Population  Crude Rate  
0                                 D1      69     1087149         6.3  
1                                 D2      14     1087149  Unreliable  
2                                 D1     424     3758097        11.3  
3                                 D2      71     3758097         1.9  
4                                 D4      19     3758097  Unreliable  
...                              ...     ...         ...         ...  
5536                              D2      15      873221  Unreliable  
5537                              D1      13      440125  Unreliable  
5538                              D1      51      426347        12.0  
5539                              D4      10      426347  Unreliable  
5540                              D1      14      172114  Unreliable  

[5541 rows x 8 columns]

#### Normalize the mortality data by population, i.e. number of deaths per 100,000 population.
#### https://handsondataviz.org/normalize.html


In [61]:
Mortality_Rate['Norm_Deaths'] = (Mortality_Rate['Deaths']/Mortality_Rate['Population'])*100000
Mortality_Rate.head(4)

Notes              County  County Code  \
0    NaN  Autauga County, AL         1001   
1    NaN  Autauga County, AL         1001   
2    NaN  Baldwin County, AL         1003   
3    NaN  Baldwin County, AL         1003   

                          Drug/Alcohol Induced Cause  \
0  Drug poisonings (overdose) Unintentional (X40-...   
1       Drug poisonings (overdose) Suicide (X60-X64)   
2  Drug poisonings (overdose) Unintentional (X40-...   
3       Drug poisonings (overdose) Suicide (X60-X64)   

  Drug/Alcohol Induced Cause Code  Deaths  Population  Crude Rate  Norm_Deaths  
0                              D1      69     1087149         6.3     6.346876  
1                              D2      14     1087149  Unreliable     1.287772  
2                              D1     424     3758097        11.3    11.282306  
3                              D2      71     3758097         1.9     1.889254

## Task 3 Member

#### Merge all the three datasets to create a super datafame
#### Read County Health ranking data using pandas

In [62]:
County_Health_Rank = pd.read_csv("../../../../data/County_Health_Ranking.csv")
County_Health_Rank.head()

State FIPS Code  County FIPS Code  5-digit FIPS Code State Abbreviation  \
0                1                 0               1000                 AL   
1                1                 1               1001                 AL   
2                1                 3               1003                 AL   
3                1                 5               1005                 AL   
4                1                 7               1007                 AL   

             Name  Release Year  County Ranked (Yes=1/No=0)  \
0         Alabama          2019                         NaN   
1  Autauga County          2019                         1.0   
2  Baldwin County          2019                         1.0   
3  Barbour County          2019                         1.0   
4     Bibb County          2019                         1.0   

   Premature death raw value  Premature death numerator  \
0                9917.232898                    80440.0   
1                8824.057123                      815.0   
2                7224.632160                     2827.0   
3                9586.165037                      451.0   
4               11783.543680                      445.0   

   Premature death denominator  ...  Male population 18-44 raw value  \
0                   13636816.0  ...                              NaN   
1                     156132.0  ...                              NaN   
2                     576496.0  ...                              NaN   
3                      72222.0  ...                              NaN   
4                      63653.0  ...                              NaN   

   Male population 45-64 raw value  Male population 65+ raw value  \
0                              NaN                            NaN   
1                              NaN                            NaN   
2                              NaN                            NaN   
3                              NaN                            NaN   
4                              NaN                            NaN   

   Total male population raw value  Female population 0-17 raw value  \
0                              NaN                               NaN   
1                              NaN                               NaN   
2                              NaN                               NaN   
3                              NaN                               NaN   
4                              NaN                               NaN   

   Female population 18-44 raw value  Female population 45-64 raw value  \
0                                NaN                                NaN   
1                                NaN                                NaN   
2                                NaN                                NaN   
3                                NaN                                NaN   
4                                NaN                                NaN   

   Female population 65+ raw value  Total female population raw value  \
0                              NaN                                NaN   
1                              NaN                                NaN   
2                              NaN                                NaN   
3                              NaN                                NaN   
4                              NaN                                NaN   

   Population growth raw value  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN  

[5 rows x 534 columns]

#### Merge all the three datasets to create a super datafame.
#### Read Opioid Proscription Rate data using pandas.

In [63]:
Opiod_Dispensing_Rate = pd.read_csv("../../../../data/2019-Opioid_Rate.csv",encoding= 'unicode_escape')

#### Mortality_Rate has a county column that provides more comprehensive information compared to Opiod_Dispensing_Rate county column.
#### I'll drop Opiod_Dispensing_Rate county column.

In [64]:
del Opiod_Dispensing_Rate['County']
Opiod_Dispensing_Rate.head()

State  FIPS  Opiod_Dispensing_Rate
0    AL  1001                  101.3
1    AL  1003                   67.6
2    AL  1005                   27.2
3    AL  1007                   21.0
4    AL  1009                   23.7

#### Merge all the three datasets to create a super datafame.
#### Merge Opiod_Dispensing_Rate and County_Health_Rank data frames.

In [65]:
Opiod_Dispensing_Rate_Merge_County_Health_Rank = pd.merge(County_Health_Rank,Opiod_Dispensing_Rate,right_on = "FIPS",
                                                          left_on = "5-digit FIPS Code", how = 'inner')
Opiod_Dispensing_Rate_Merge_County_Health_Rank.head()

State FIPS Code  County FIPS Code  5-digit FIPS Code State Abbreviation  \
0                1                 1               1001                 AL   
1                1                 3               1003                 AL   
2                1                 5               1005                 AL   
3                1                 7               1007                 AL   
4                1                 9               1009                 AL   

             Name  Release Year  County Ranked (Yes=1/No=0)  \
0  Autauga County          2019                         1.0   
1  Baldwin County          2019                         1.0   
2  Barbour County          2019                         1.0   
3     Bibb County          2019                         1.0   
4   Blount County          2019                         1.0   

   Premature death raw value  Premature death numerator  \
0                8824.057123                      815.0   
1                7224.632160                     2827.0   
2                9586.165037                      451.0   
3               11783.543680                      445.0   
4               10908.101820                     1050.0   

   Premature death denominator  ...  Total male population raw value  \
0                     156132.0  ...                              NaN   
1                     576496.0  ...                              NaN   
2                      72222.0  ...                              NaN   
3                      63653.0  ...                              NaN   
4                     161107.0  ...                              NaN   

   Female population 0-17 raw value  Female population 18-44 raw value  \
0                               NaN                                NaN   
1                               NaN                                NaN   
2                               NaN                                NaN   
3                               NaN                                NaN   
4                               NaN                                NaN   

   Female population 45-64 raw value  Female population 65+ raw value  \
0                                NaN                              NaN   
1                                NaN                              NaN   
2                                NaN                              NaN   
3                                NaN                              NaN   
4                                NaN                              NaN   

   Total female population raw value  Population growth raw value  State  \
0                                NaN                          NaN     AL   
1                                NaN                          NaN     AL   
2                                NaN                          NaN     AL   
3                                NaN                          NaN     AL   
4                                NaN                          NaN     AL   

   FIPS  Opiod_Dispensing_Rate  
0  1001                  101.3  
1  1003                   67.6  
2  1005                   27.2  
3  1007                   21.0  
4  1009                   23.7  

[5 rows x 537 columns]

#### Merge all the three datasets to create a super datafame
#### Merge result of above merge and Mortality_Rate data frame

In [66]:
Merged_df = pd.merge(Mortality_Rate,Opiod_Dispensing_Rate_Merge_County_Health_Rank, right_on = "FIPS",
                                                          left_on = "County Code", how = 'inner')
Merged_df.head()

Notes              County  County Code  \
0    NaN  Autauga County, AL         1001   
1    NaN  Autauga County, AL         1001   
2    NaN  Baldwin County, AL         1003   
3    NaN  Baldwin County, AL         1003   
4    NaN  Baldwin County, AL         1003   

                          Drug/Alcohol Induced Cause  \
0  Drug poisonings (overdose) Unintentional (X40-...   
1       Drug poisonings (overdose) Suicide (X60-X64)   
2  Drug poisonings (overdose) Unintentional (X40-...   
3       Drug poisonings (overdose) Suicide (X60-X64)   
4  Drug poisonings (overdose) Undetermined (Y10-Y14)   

  Drug/Alcohol Induced Cause Code  Deaths  Population  Crude Rate  \
0                              D1      69     1087149         6.3   
1                              D2      14     1087149  Unreliable   
2                              D1     424     3758097        11.3   
3                              D2      71     3758097         1.9   
4                              D4      19     3758097  Unreliable   

   Norm_Deaths  State FIPS Code  ...  Total male population raw value  \
0     6.346876                1  ...                              NaN   
1     1.287772                1  ...                              NaN   
2    11.282306                1  ...                              NaN   
3     1.889254                1  ...                              NaN   
4     0.505575                1  ...                              NaN   

   Female population 0-17 raw value Female population 18-44 raw value  \
0                               NaN                               NaN   
1                               NaN                               NaN   
2                               NaN                               NaN   
3                               NaN                               NaN   
4                               NaN                               NaN   

  Female population 45-64 raw value  Female population 65+ raw value  \
0                               NaN                              NaN   
1                               NaN                              NaN   
2                               NaN                              NaN   
3                               NaN                              NaN   
4                               NaN                              NaN   

   Total female population raw value  Population growth raw value  State  \
0                                NaN                          NaN     AL   
1                                NaN                          NaN     AL   
2                                NaN                          NaN     AL   
3                                NaN                          NaN     AL   
4                                NaN                          NaN     AL   

   FIPS  Opiod_Dispensing_Rate  
0  1001                  101.3  
1  1001                  101.3  
2  1003                   67.6  
3  1003                   67.6  
4  1003                   67.6  

[5 rows x 546 columns]

In [67]:
Merged_df.shape

(5538, 546)

In [68]:
super_df = pd.read_csv('../../../../data/super_df.csv')
super_df.head()

County State  County Code  Population  Deaths  Norm_Deaths  \
0  Abbeville County, SC    SC        45001      535389      38          8.0   
1     Acadia Parish, LA    LA        22001     1279727     237         19.0   
2   Accomack County, VA    VA        51001      726189      73         11.0   
3        Ada County, ID    ID        16001     8083452     988         13.0   
4      Adair County, KY    KY        21001      387950      43         12.0   

   State FIPS Code  County FIPS Code  5-digit FIPS Code State Abbreviation  \
0               45                 1              45001                 SC   
1               22                 1              22001                 LA   
2               51                 1              51001                 VA   
3               16                 1              16001                 ID   
4               21                 1              21001                 KY   

   ... Male population 65+ raw value  Total male population raw value  \
0  ...                           NaN                              NaN   
1  ...                           NaN                              NaN   
2  ...                           NaN                              NaN   
3  ...                           NaN                              NaN   
4  ...                           NaN                              NaN   

   Female population 0-17 raw value  Female population 18-44 raw value  \
0                               NaN                                NaN   
1                               NaN                                NaN   
2                               NaN                                NaN   
3                               NaN                                NaN   
4                               NaN                                NaN   

   Female population 45-64 raw value  Female population 65+ raw value  \
0                                NaN                              NaN   
1                                NaN                              NaN   
2                                NaN                              NaN   
3                                NaN                              NaN   
4                                NaN                              NaN   

   Total female population raw value  Population growth raw value   FIPS  \
0                                NaN                          NaN  45001   
1                                NaN                          NaN  22001   
2                                NaN                          NaN  51001   
3                                NaN                          NaN  16001   
4                                NaN                          NaN  21001   

   Opiod_Dispensing_Rate  
0                   32.4  
1                   32.2  
2                   19.4  
3                   60.8  
4                   71.3  

[5 rows x 542 columns]

#### Now, I'll check if I am selecting all columns of super_df


In [69]:
super_df_cols = super_df.columns
Merged_df_cols = Merged_df.columns
# columns in super_df and not Merged_df
set(super_df_cols) - set(Merged_df_cols)

set()

In [70]:
# columns in Merged_df and not super_df 
set(Merged_df_cols) - set(super_df_cols) 

{'Crude Rate',
 'Drug/Alcohol Induced Cause',
 'Drug/Alcohol Induced Cause Code',
 'Notes'}

#### Save merged df to 'csv'.


In [71]:
Merged_df = Merged_df.set_index('County')
Merged_df.to_csv('Merged_DataFrame.csv')

#### M3.2 Identify counties and states (top 10) with the highest opioid mortality rates. (20 pts)
#### Use mean and median for counties within states to compare (for the state level).
#### Describe your intution on why the rates are high in these states and counties.

In [72]:
# Identify counties with the highest opioid mortality rates.
Merged_df['Norm_Deaths'].sort_values(ascending = False)[:10]

County
Rio Arriba County, NM    56.607603
Wyoming County, WV       52.822994
McDowell County, WV      52.692188
Boone County, WV         45.350321
Powell County, KY        44.494298
Cabell County, WV        43.392627
Floyd County, KY         41.968516
Logan County, WV         41.441748
Mingo County, WV         40.929884
Baltimore city, MD       40.721996
Name: Norm_Deaths, dtype: float64

#### My intuition:
#### Two states (West Virgina and Kentucky) has two or more counties in this list. 
#### This indicates that opioid epedimic could be a serious issue across most counties in these states 

#### Identify states with the highest opioid mortality rates.

In [73]:

Merged_df.groupby('State')['Norm_Deaths'].sum().sort_values(ascending = False)[:10]

State
KY    2208.942202
TN    1615.042707
TX    1470.977963
NC    1381.482012
VA    1377.872032
OH    1358.397103
GA    1328.566852
IN    1170.918048
WV    1168.861923
MO    1107.341102
Name: Norm_Deaths, dtype: float64

#### Calculate mean and median death rate at state level.


In [74]:
State_Mean = Merged_df.groupby('State')['Norm_Deaths'].mean().reset_index()
State_Mean.columns = ['State','Norm_Deaths_Mean']
State_Median = Merged_df.groupby('State')['Norm_Deaths'].median().reset_index()
State_Median.columns = ['State','Norm_Deaths_Median']
State_Info = pd.merge(State_Median,State_Mean, on = 'State', how = 'inner')
State_Info.head()

State  Norm_Deaths_Median  Norm_Deaths_Mean
0    AK            5.136560          5.917706
1    AL            2.643914          4.432717
2    AR            3.495702          4.184923
3    AZ            2.235898          4.794026
4    CA            1.704058          4.338118

#### Merge State_Info data amd Merged_df to compare county death rate with overall state death rate.


In [75]:
Merged_df = pd.merge(State_Info,Merged_df,on = 'State', how = 'inner')
Merged_df.head()

State  Norm_Deaths_Median  Norm_Deaths_Mean  Notes  County Code  \
0    AK             5.13656          5.917706    NaN         2016   
1    AK             5.13656          5.917706    NaN         2020   
2    AK             5.13656          5.917706    NaN         2020   
3    AK             5.13656          5.917706    NaN         2020   
4    AK             5.13656          5.917706    NaN         2020   

                          Drug/Alcohol Induced Cause  \
0  Drug poisonings (overdose) Unintentional (X40-...   
1  Drug poisonings (overdose) Unintentional (X40-...   
2       Drug poisonings (overdose) Suicide (X60-X64)   
3  Drug poisonings (overdose) Undetermined (Y10-Y14)   
4                      All other drug-induced causes   

  Drug/Alcohol Induced Cause Code  Deaths  Population  Crude Rate  ...  \
0                              D1      12      117650  Unreliable  ...   
1                              D1     803     5960535        13.5  ...   
2                              D2     112     5960535         1.9  ...   
3                              D4      47     5960535         0.8  ...   
4                              D9      62     5960535         1.0  ...   

   Male population 65+ raw value  Total male population raw value  \
0                            NaN                              NaN   
1                            NaN                              NaN   
2                            NaN                              NaN   
3                            NaN                              NaN   
4                            NaN                              NaN   

   Female population 0-17 raw value  Female population 18-44 raw value  \
0                               NaN                                NaN   
1                               NaN                                NaN   
2                               NaN                                NaN   
3                               NaN                                NaN   
4                               NaN                                NaN   

  Female population 45-64 raw value Female population 65+ raw value  \
0                               NaN                             NaN   
1                               NaN                             NaN   
2                               NaN                             NaN   
3                               NaN                             NaN   
4                               NaN                             NaN   

   Total female population raw value  Population growth raw value  FIPS  \
0                                NaN                          NaN  2016   
1                                NaN                          NaN  2020   
2                                NaN                          NaN  2020   
3                                NaN                          NaN  2020   
4                                NaN                          NaN  2020   

   Opiod_Dispensing_Rate  
0                    2.5  
1                   56.0  
2                   56.0  
3                   56.0  
4                   56.0  

[5 rows x 547 columns]

#### Find number of counties in each sate.


In [76]:
County_Count = Merged_df.groupby('State').count()['FIPS'].reset_index()
County_Count.columns = ['State','County Count']
County_Count.head()

State  County Count
0    AK            28
1    AL           123
2    AR           136
3    AZ            48
4    CA           193

#### Find counties with death rate above state mean death rate.


In [77]:
Merged_df_above_mean = Merged_df[Merged_df.Norm_Deaths>= Merged_df.Norm_Deaths_Mean ]
County_Count_Above_Mean = Merged_df_above_mean.groupby('State').count()['FIPS'].reset_index()
County_Count_Above_Mean.columns = ['State','Counties Above Mean State Death Rate']
County_Count_Above_Mean.head()

State  Counties Above Mean State Death Rate
0    AK                                    13
1    AL                                    52
2    AR                                    56
3    AZ                                    15
4    CA                                    57

#### Find counties with death rate above state median death rate.


In [78]:
Merged_df_above_median = Merged_df[Merged_df.Norm_Deaths>= Merged_df.Norm_Deaths_Median ]
County_Count_Above_Median = Merged_df_above_median.groupby('State').count()['FIPS'].reset_index()
County_Count_Above_Median.columns = ['State','Counties Above Median State Death Rate']
County_Count_Above_Median.head()

State  Counties Above Median State Death Rate
0    AK                                      14
1    AL                                      62
2    AR                                      68
3    AZ                                      24
4    CA                                      97

#### merge three above data frames:
#### I did not know how to merge four data frames at the same time.
#### https://stackoverflow.com/questions/23668427/pandas-three-way-joining-multiple-dataframes-on-columns


In [79]:
import functools as ft
dfs = [County_Count,County_Count_Above_Mean,County_Count_Above_Median,State_Info]
State_Info_final = ft.reduce(lambda left, right: pd.merge(left, right, on='State'), dfs)
State_Info_final.head()

State  County Count  Counties Above Mean State Death Rate  \
0    AK            28                                    13   
1    AL           123                                    52   
2    AR           136                                    56   
3    AZ            48                                    15   
4    CA           193                                    57   

   Counties Above Median State Death Rate  Norm_Deaths_Median  \
0                                      14            5.136560   
1                                      62            2.643914   
2                                      68            3.495702   
3                                      24            2.235898   
4                                      97            1.704058   

   Norm_Deaths_Mean  
0          5.917706  
1          4.432717  
2          4.184923  
3          4.794026  
4          4.338118

#### States with most counties above state mean death rate:


In [80]:
State_Info_final[ State_Info_final['Counties Above Mean State Death Rate'] / State_Info_final['County Count'] > 0.5]

State  County Count  Counties Above Mean State Death Rate  \
36    OK           129                                    65   

    Counties Above Median State Death Rate  Norm_Deaths_Median  \
36                                      65            7.812354   

    Norm_Deaths_Mean  
36          7.743447

#### Only in few sates most counties death rate is greater than state mean death rate.
#### States with most counties above state median death rate:

In [81]:
State_Info_final[ State_Info_final['Counties Above Median State Death Rate'] / State_Info_final['County Count'] > 0.5]

State  County Count  Counties Above Mean State Death Rate  \
1     AL           123                                    52   
4     CA           193                                    57   
7     DC             5                                     1   
9     FL           189                                    65   
10    GA           225                                   108   
11    HI            15                                     5   
14    IL           175                                    75   
17    KY           177                                    80   
20    MD            81                                    24   
21    ME            39                                    16   
22    MI           199                                    77   
25    MS           103                                    39   
26    MT            49                                    19   
27    NC           237                                    95   
28    ND            21                                    10   
30    NH            33                                    10   
31    NJ            81                                    21   
32    NM            69                                    29   
35    OH           223                                    80   
36    OK           129                                    65   
37    OR            95                                    37   
40    SC            99                                    42   
41    SD            21                                     9   
42    TN           215                                    90   
43    TX           343                                   153   
45    VA           225                                    96   
46    VT            35                                    13   
50    WY            29                                    13   

    Counties Above Median State Death Rate  Norm_Deaths_Median  \
1                                       62            2.643914   
4                                       97            1.704058   
7                                        3            1.779658   
9                                       95            2.354028   
10                                     113            5.610800   
11                                       8            1.769736   
14                                      88            3.473298   
17                                      89           10.957471   
20                                      41            3.076737   
21                                      20            2.555786   
22                                     100            2.953490   
25                                      52            4.575798   
26                                      25            4.425072   
27                                     119            2.261913   
28                                      11            3.501337   
30                                      17            2.064830   
31                                      41            1.079351   
32                                      35            3.984921   
35                                     112            1.684403   
36                                      65            7.812354   
37                                      48            2.995434   
40                                      50            1.678534   
41                                      11            2.456936   
42                                     108            3.504093   
43                                     172            3.155689   
45                                     113            4.648058   
46                                      18            2.168502   
50                                      15            5.175309   

    Norm_Deaths_Mean  
1           4.432717  
4           4.338118  
7           4.328437  
9           5.194197  
10          5.904742  
11          2.878887  
14          4.729436  
17         12.479899  
20          5.300048  
2

#### In 25 states most counties have death rates larger than state median death rate.
#### From top 10 sates with highest death rate 
#### KY    2208.942202
#### TN    1615.042707
#### TX    1470.977963
#### NC    1381.482012
#### VA    1377.872032
#### OH    1358.397103
#### GA    1328.566852
#### IN    1170.918048
#### WV    1168.861923
#### MO    1107.341102

In [82]:
Top10_State_Death_Rate = list(Merged_df.groupby('State')['Norm_Deaths'].sum().sort_values(ascending = False)[:10].index)
Most_Counties_Above_Median = list(State_Info_final[ State_Info_final['Counties Above Median State Death Rate'] / State_Info_final['County Count'] > 0.5].State)
Most_Counties_Above_Mean = list(State_Info_final[ State_Info_final['Counties Above Mean State Death Rate'] / State_Info_final['County Count'] > 0.5].State)

#### Does most counties in states with highest death rates have death rates above state mean death rate?  


In [83]:
set(Top10_State_Death_Rate) - set(Most_Counties_Above_Mean)

{'GA', 'IN', 'KY', 'MO', 'NC', 'OH', 'TN', 'TX', 'VA', 'WV'}

#### From top 10 sates with highest death rate, non of them is in the list of states with most counties having death rate larger than state mean death rate. This indicates that the issue is more local and is specific to some counties in state rather than all counties in a state.

#### Does most counties in states with highest death rates have death rates above state median death rate?  


In [84]:
set(Top10_State_Death_Rate) - set(Most_Counties_Above_Median)

{'IN', 'MO', 'WV'}

#### From top 10 sates with highest death rate, only 5 of them (non of the top 3 states) are in the list of states with most counties having death rate larger than state median death rate. This indicates that the issue can still be quite local. It might be specific to some counties in state rather than all counties in a state. Some counties might be higly deprived compared to other counties in such sates.